packages/rspeedy/core/src/plugins/generator.plugin.ts

```diff
@@ -0,0 +1,20 @@
+// Copyright 2024 The Lynx Authors. All rights reserved.
+// Licensed under the Apache License Version 2.0 that can be found in the
+// LICENSE file in the root directory of this source tree.
+import type { RsbuildPlugin } from '@rsbuild/core'
+
+export function pluginGenerator(): RsbuildPlugin {
+  return {
+    name: 'lynx:rsbuild:generator',
+    async setup(api) {
+      api.modifyBundlerChain(chain => {
+        // Avoid generating `JSON.parse()` for a JSON file which has more than 20 characters as it increases the bundle size of `main-thread.js`.
+        // For more detail, see https://github.com/webpack/webpack/issues/19319
+        chain.module
+          .generator.set('json', {
+            JSONParse: false,
```

Comment:
I think we should only set `JSONParse: false` on the main-thread (adding `layer` in this rule). Since the `background.js` is not compiled to bytecode and it should be faster to use `JSON.parse`.

Maybe you should add this rule in `@lynx-js/react-rsbuild-plugin`.

Reply:
Done.